### PR TITLE
Coordinator ddata state store with persistent remember entities

### PR DESCRIPTION
**Pull request by johanandren**
References #28961

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberShards.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      sender() ! RememberEntitiesCoordinatorStore.RememberedShards(shards)
+
+    case RememberEntitiesCoordinatorStore.AddShard(shardId: ShardId) =>
+      persistAsync(shardId) { shardId =>
+        shards += shardId
+        sender() ! RememberEntitiesCoordinatorStore.UpdateDone(shardId)
+      }
+  }
+
+}
```
Comment (johanandren) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberShards.scala:48]: This does not cover how we'll migrate.

One idea I had was to make a persistent actor that uses the same id as a coordinator and accepts the events of that but only hangs on to the shard ids. Not sure it's a good idea thought.

Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberShards.scala:48]: An alternative could be to have a migration version of the persistence based coordinator, that does the same, and then sends the full list of shards to the store when it has recovered. That way we wouldn't have to mix that logic into the regular event sourced store for the coordinator..

**Comment by johanandren** [comment]
Multi jvm test added for ddata store mode and event sourced remember entities store, updated tests a bit since this requires that a journal is set up if either persistent state store or event sourced remember entitites.

I think I'll stop here for this week.

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberShards.scala (hunk 1 above)
Comment (chbatey) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberShards.scala:48]: As discussed let's merge this and i'll work on the migration

**Review by chbatey — APPROVED** [review]

**Pull request merged**